Makefile

```diff
@@ -54,19 +72,21 @@ else
       $(warning OpenMP not found, skipping OpenMP support)
     endif
   else
-    # Check for OpenMP support in GCC or Clang on Linux
-    ifeq ($(shell echo | $(CC) -fopenmp -x c -E - > /dev/null 2>&1; echo $$?), 0)
-      CFLAGS += -fopenmp -DOMP
-      LDLIBS += -lgomp
-      $(info OpenMP found, compiling with OpenMP support)
-    else
-      $(warning OpenMP not found, skipping OpenMP support)
+    ifneq ($(OS), Windows_NT)
+      # Check for OpenMP support in GCC or Clang on Linux
+      ifeq ($(shell echo | $(CC) -fopenmp -x c -E - > /dev/null 2>&1; echo $$?), 0)
+        CFLAGS += -fopenmp -DOMP
+        LDLIBS += -lgomp
+        $(info OpenMP found, compiling with OpenMP support)
+      else
+        $(warning OpenMP not found, skipping OpenMP support)
+      endif
     endif
   endif
 endif
 
-# PHONY means these targets will always be executed
-.PHONY: all train_gpt2 test_gpt2 train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu
```

Comment:
.PHONY is very convenient. It rebuilds the files even if they exists. Why is it removed? I would like it back..

Reply:
Ah - was trying to avoid that! My preference is know if a target has been built and to not rebuild it. I've normally seen .PHONY only have special commands and not targets. What's the rationale for wanting to build something that's already successfully been built? And no problem putting it back. Thought it was a mistake...no worries...

Reply:
The comment above PHONY is misleading, perhaps rewrite it. PHONY means that Make will not check if the file exists before deciding to build it, which makes it seem like they are always executed which is why the comment is partly correct. But since you introduced DEFAULT, there is a little bit of confusion.. By default, without the DEFAULT, the first recipe definition is used. In this case it is "all".

Reply:
Yes, that confused me.